### PR TITLE
Add possbility to define one filter for several layers

### PR DESCRIPTION
**Pull request by troopa81**
## Description

Sometimes, we need to define the same filter for every selected layer when making a WMS request ( when accessing to user specific data for instance). For now we have to write : 

```
&FILTER=layer1:"column1" = 'value1';layer2:"column1" = 'value1';layer3:"column1" = 'value1'
```

This PR adds the possibility to simplify the writing with this syntax:

```
&FILTER=layer1,layer2,layer3:"column1" = 'value1'
```

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/services/wms/qgswmsparameters.cpp [hunk 1]
```diff
@@ -1329,15 +1329,19 @@ namespace QgsWms
...
+        // filter format: "LayerName,LayerName2:filterString;LayerName3:filterString2;..."
         // several filters can be defined for one layer
         const QStringList splits = f.split( ':' );
         if ( splits.size() == 2 )
         {
-          QgsWmsParametersFilter filter;
-          filter.mFilter = splits[1];
-          filter.mType = QgsWmsParametersFilter::SQL;
-          filters.insert( splits[0], filter );
+          QStringList layers = splits[0].contains( ',' ) ? splits[0].split( ',' ) : QStringList() << splits[0];
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsparameters.cpp]:
```suggestion
          const QStringList layers = splits[0].contains( ',' ) ? splits[0].split( ',' ) : QStringList() << splits[0];
```

Comment (m-kuhn) [reply src/server/services/wms/qgswmsparameters.cpp]: Is the `?` required?

I would imagine a simple `const QStringList layer = splits.at( 0 ).split( ',' );` would produce the same, no?

Comment (troopa81) [reply src/server/services/wms/qgswmsparameters.cpp]: you right, I made the modifications

File: src/server/services/wms/qgswmsparameters.cpp [hunk 2]
```diff
@@ -1329,15 +1329,19 @@ namespace QgsWms
...
         // several filters can be defined for one layer
         const QStringList splits = f.split( ':' );
         if ( splits.size() == 2 )
         {
-          QgsWmsParametersFilter filter;
-          filter.mFilter = splits[1];
-          filter.mType = QgsWmsParametersFilter::SQL;
-          filters.insert( splits[0], filter );
+          const QStringList layers = splits[0].split( ',' );
+          for ( QString layer : layers )
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsparameters.cpp]: I think that `layer` should be a const reference in this case.

**Comment by pblottiere** [comment]
Hi @troopa81,

Just a minor comment, otherwise the code looks good :+1:.

Can you add a unit test with an invalid expression for one of the layer? Just to check that an exception is well raised in this case too.

And I think that the mask should be updated to please Travis :).

Thanks!

**Comment by pblottiere** [comment]
Thanks @troopa81, good work :+1:!

**Pull request merged**